inference-engine/tools/compile_tool/main.cpp

```diff
@@ -343,49 +395,83 @@ static void processPrecisions(InferenceEngine::CNNNetwork &network,
         }
     }
 
-    if (!outputs_precision.empty()) {
-        auto precision = getOutputPrecision(outputs_precision);
-        for (auto &&layer : network.getOutputsInfo()) {
+    if (!FLAGS_op.empty()) {
+        auto precision = getPrecision(FLAGS_op);
+        for (auto&& layer : network.getOutputsInfo()) {
             const auto layerPrecision = layer.second->getPrecision();
             if (isFloat(layerPrecision) && isFloat(precision)) {
```

Comment:
Still isFloat check, it can be a problem for ICNEt

Reply:
The logic is:

* If user provides only `-op` option (global precision set), we do the checks to avoid issues with type casting.
* If user provides `-iop "output:FP32"` option we don't perform any checks and assume user knows what he/she is doing.

Reply:
Updated. Now all checks for precision was removed and its applicability/support is under user/plugin responsibility.